### PR TITLE
WIP: Implement concrete classes (Don't review)

### PR DESCRIPTION
**Pull request by UncleGrandpa925**
In this PR I'm modifying & creating 

- [x] Lesson
- [x] Module
- [x] Event
- [x] Parser Inteface
- [x] Command Abstract Class

- [ ] Model Interface

- [ ] ModelManager
- [ ] Logic Interface

- [ ] Logic Manager

### CODE REVIEW

File: src/main/java/seedu/address/model/lesson/Location.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public String toString() {
+        return location;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof Name // instanceof handles nulls
```
Comment (simonteozw) [inline src/main/java/seedu/address/model/lesson/Location.java]: Can I check why we are checking if other is instanceof Name? We should only check if other is also a Location right

File: src/main/java/seedu/address/model/task/Location.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Override
+    public String toString() {
+        return location;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
+            || (other instanceof Name // instanceof handles nulls
```
Comment (simonteozw) [inline src/main/java/seedu/address/model/task/Location.java]: Another instance here. I think should be other instanceof Location right?

**Review by simonteozw — COMMENTED** [review]
Nice code Long! Think there are 1 or 2 minor errors. There is an Address class in the 'contact' directory currently. Would it make sense to use that Address to store all kinds of location data? Just thinking.

File: src/main/java/seedu/address/model/lesson/Location.java (hunk 1 above)
Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/lesson/Location.java]: My bad😢

File: src/main/java/seedu/address/model/task/Location.java (hunk 2 above)
Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/task/Location.java]: True

File: src/main/java/seedu/address/model/task/Location.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import seedu.address.model.commons.Name;
+
+public class Location {
+    public final String location;
+
+    public Location(String location) {
+        requireNonNull(location);
+        this.location = location;
+    }
+
```
Comment (simonteozw) [inline src/main/java/seedu/address/model/task/Location.java]: Should we use Address Class for Task as well? I am unclear if there is a difference between Location and Address class class haha

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -58,7 +58,7 @@ public void parseIndex_validInput_success() throws Exception {
 
     @Test
     public void parseName_null_throwsNullPointerException() {
-        assertThrows(NullPointerException.class, () -> ParserUtil.parseName((String) null));
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseName(null));
```
Comment (simonteozw) [inline src/test/java/seedu/address/logic/parser/ParserUtilTest.java:61]: Just curious, why did you remove the `(String)` typecasting?

**Review by simonteozw — COMMENTED** [review]
Thanks for the fix Long!

File: src/main/java/seedu/address/model/task/Location.java (hunk 3 above)
Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/task/Location.java]: Hmm I have removed this class already?

Comment (UncleGrandpa925) [reply src/main/java/seedu/address/model/task/Location.java]: Ah okay not yet, my bad

File: src/test/java/seedu/address/logic/parser/ParserUtilTest.java (hunk 4 above)
Comment (UncleGrandpa925) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:61]: Can you take a look at my new PR?

Comment (UncleGrandpa925) [reply src/test/java/seedu/address/logic/parser/ParserUtilTest.java:61]: If that PR is good then can merge it first, so that there won't be too many commits in a PR
